site/public/js/forum.js

```diff
@@ -377,7 +377,16 @@ function socketNewOrEditThreadHandler(thread_id, edit=false) {
                         const last = last_bookmarked.length === 0 ? last_announcement : last_bookmarked;
 
                         if (last.length === 0) {
-                            $(new_thread).insertBefore($('.thread_box_link').first()).hide().fadeIn('slow');
+                            if ($(new_thread).find('.thread-announcement-expiring').length === 1) {
+                                $(new_thread).insertBefore($('.thread_box_link').first()).hide().fadeIn('slow');
+                            }
+                            else {
+                                let spot_after_announcements = $('.thread_box_link').first();
```

Comment:
This variable doesn't appear to be reassigned after the initial declaration, so it should be defined as a constant.

```suggestion
                                const spot_after_announcements = $('.thread_box_link').first();
```